### ISSUE TITLE
Not recognized as monospaced font

### ISSUE THREAD

**Issue by tingtron** [damianvila/font-cpc464#1]
Buenos dias!
Thank you for sharing this full-featured Amstrad font - even with the dancing person.

There is one big problem for using this font in monospace situation. Many applications, which expect or approximate monospace fonts work OK (Terminal, WordPad for Windows). But there was a problem with SDL-related font libraries. It treated some characters (like 0,4,W,M) wider than others, by 1 pixel at 20 size. But it was enough to through off alignment, and it was no longer monospace.

Good news is that the fix is simple. As said in the [FontForge FAQ](https://fontforge.org/docs/faq.html),

> **How do I mark a font as monospaced?**
>- You don’t. Just insure that **all** the glyphs in the font have the same width and then FontForge will automatically mark it as monospaced for you. (If you mark it as monospaced incorrectly some rasterizers will give strange results).
>   ...
>   When I say “all glyphs” I really mean all glyphs. Even glyphs which unicode says should be 0 width must have the same width as everything else.

It turns out that the very last "no glyph" was the culprit having only 400 width instead of the 800 as the em-size and the rest of the glyphs. So extending the width (and optionally re-centering the image) did the trick. Now the SDL-related font API is happy, showing all characters in the same width.

![image](https://user-images.githubusercontent.com/2383820/213976897-749adce9-162a-4aed-82c5-110cfae1784c.png)

For comparison, before the fix:
![image](https://user-images.githubusercontent.com/2383820/213977626-cac55fe2-5267-4993-9e5e-765decaf23a9.png)
After the fix:
![image](https://user-images.githubusercontent.com/2383820/213977652-d4860aa0-c24e-4b83-99b4-495854a7f173.png)

**Comment by damianvila** [comment]
Thank for the info. I'l try to fix it asap. This font was created with the Glyphs app for Mac, in the 30 day trial (now expired). I've never used Font Forge, so it may take a while to correct it. When ready, I'll create a new version. Cheers.

**Comment by damianvila** [comment]
@tingtron Can you check the Pre-release and tell me if the error is fixed? If it's ok I'll change it to a release. Thanks.

**Comment by tingtron** [comment]
Thank you for taking care of this. It looks like the updated font works well in my tests as above. And I see the last glyph being the same widths as the others. So this issue can be considered closed.

For the future, you might consider this minor issue reported when exporting the font when Generating the resulting TTF:
![image](https://user-images.githubusercontent.com/2383820/216238373-bcc4244a-6d99-4503-9c2e-cc24401f8e9a.png)
It does not cause any issues in my tests. It is not reported among other Validations, just when Generating.

I can see how 800 is convenient for manual creation of an 8-pixel font. So such normalization to maybe 1024 would probably make sense for the case of automated font outlining from pixels. 
Or maybe a version of this font scaled to 1024. In FontForge there is Transform/Scale/Uniform ... by 128% will produce 1024. Then the font metrics (ascent, descent etc.) need to be adjusted too.

FontForge looks pretty solid and convenient, similar to InkScape.

**Comment by damianvila** [comment]
Perfect! Yes, I saw the error. But as I've said, I developed these fonts using a Mac App (Glyphs) that has its own quirks. I didn't want to modify the files much, at the risk of ruining the font (I don't have that app anymore to edit the original files).
As a Mac user, the interface of FontForge is a bit weird, but I'll try to get used to it, since I want to keep making these kind of fonts. And I want to be able to edit the fonts in the future, if another problem appears. If I correct these issues (when I feel comfortable using FontForge), I'll update the release. Thanks for the comments. Cheers!
